internal/services/sources_service.go

```diff
@@ -42,3 +44,104 @@ func ListSources(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+func GetAWSAccountIdentity(w http.ResponseWriter, r *http.Request) {
+	sourceId := chi.URLParam(r, "ID")
+
+	sourcesClient, err := clients.GetSourcesClient(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewClientError(r.Context(), err))
+		return
+	}
+
+	authentication, err := sourcesClient.GetAuthentication(r.Context(), sourceId)
+	if err != nil {
+		renderError(w, r, payloads.NewClientError(r.Context(), err))
+		return
+	}
+
+	if typeErr := authentication.MustBe(models.ProviderTypeAWS); typeErr != nil {
+		renderError(w, r, payloads.NewClientError(r.Context(), typeErr))
+		return
+	}
+
+	uploadInfo, err := getAWSUploadInfo(r.Context(), authentication)
+	if err != nil {
+		renderError(w, r, payloads.NewClientError(r.Context(), err))
+		return
+	}
+
+	if err := render.Render(w, r, payloads.NewAccountIdentityResponse(uploadInfo)); err != nil {
+		renderError(w, r, payloads.NewRenderError(r.Context(), "unable to render account id", err))
+		return
+	}
+}
+
+func GetSourceUploadInfo(w http.ResponseWriter, r *http.Request) {
+	sourceId := chi.URLParam(r, "ID")
```

Comment:
Why are we getting a source id as a parameter? 
I suggest we fetch all the sources from sources, loop over them, get the information from each source and send all this information once to the frontend. 
That way, we could solve the bug we encounter, avoid back and forth between the browser and our backend, and also we can filter out empty sources (which are valid according to sources)

Reply:
I think that the endpoint is used by image builder and this PR was refactoring of existing one in order to bring Azure. So that is why it exists.

Now, the new UI source step check using it is another story and we should probably introduce a new one that can be more resilient. But I would love to hear from @ezr-ondrej when he gets to this.

Reply:
yeah, I have kept it per one source, because it makes more sense in the API contract, but to fit our needs it would be better to have that for all sources, yes :)

Reply:
Just for the context: we already talked about this and agreed that we might need another endpoint. That was the last week when you were away. Now that I have little bit more info about how it works, I think we should consider maybe only frontend fix, in case this page turns out to be temporary. Let’s be just careful with that because history shows temporary solutions tend to stick :-)